sol/contracts/ics23Ops.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.2;
+
+import {LeafOp, InnerOp, PROOFS_PROTO_GLOBAL_ENUMS, ProofSpec} from "./proofs.sol";
+import {ProtoBufRuntime} from "./ProtoBufRuntime.sol";
+import {SafeCast} from "OpenZeppelin/openzeppelin-contracts@4.2.0/contracts/utils/math/SafeCast.sol";
+import {Math} from "OpenZeppelin/openzeppelin-contracts@4.2.0/contracts/utils/math/Math.sol";
+import {BytesLib} from "GNSPS/solidity-bytes-utils@0.8.0/contracts/BytesLib.sol";
+
+library Ops {
+
+    // LeafOp operations
+    function applyOp(LeafOp.Data memory leafOp, bytes memory key, bytes memory value) internal pure returns(bytes memory) {
+        require(key.length > 0); // dev: Leaf op needs key
+        require(value.length > 0); // dev: Leaf op needs value
+        bytes memory pKey = prepareLeafData(leafOp.prehash_key, leafOp.length, key);
+        bytes memory pValue = prepareLeafData(leafOp.prehash_value, leafOp.length, value);
+        bytes memory data = bytes.concat(leafOp.prefix, pKey, pValue);
```

Comment:
`bytes.concat` [comes with 0.8.4](https://docs.soliditylang.org/en/v0.8.4/types.html#bytes-concat-function). This code doesn't compile if you use 0.8.2 compiler.

I'd replace it with `abi.encodePacked` for now.